### PR TITLE
Add dataset improvements in filter and metadata panel

### PR DESCRIPTION
**Pull request by ptbrowne**
Improvements on the add dataset functionality.

- refactor: Extract out make search query
- refactor: Use enum for SearchCubeFilterType
- fix: Use correct value for TemporalDimension
- fix: Use correct component iri when tweaking iri to be joinBy
- feat: ControlTabField handle multi components field
- fix: Missing localized string
- wip fix nan
- feat: Show multiple for vertical axis in case of combo chart + show merged temporal dimension according to time unit
- feat: Use chart spec to known where are the iri attributes
- feat: Show correct dimension label in metadata panel

### CODE REVIEW

File: app/rdf/query-search.ts [hunk 1]
```diff
@@ -97,7 +97,110 @@ export const searchCubes = async ({
...
+  const scoresQuery = mkScoresQuery(
+    locale,
+    filters,
+    creatorValues,
+    themeValues,
+    includeDrafts,
+    query
+  ); // HOTFIX WRT Stardog v9.2.1 bug see https://control.vshn.net/tickets/sbar-1066
+
+  console.log(scoresQuery.build());
```
Comment (bprusinowski) [inline app/rdf/query-search.ts:110]: 🔍

File: app/graphql/schema.graphql [hunk 2]
```diff
@@ -318,8 +318,15 @@ type DataCubeOrganization {
   label: String
 }
 
+enum SearchCubeFilterType {
+  TemporalDimension
+  DataCubeTheme
+  DataCubeOrganization
+  DataCubeAbout
+}
+
```
Comment (bprusinowski) [inline app/graphql/schema.graphql:327]: 👍

File: app/configurator/components/chart-controls/control-tab.tsx [hunk 3]
```diff
@@ -84,9 +86,10 @@ const getLabels = (
   switch (value) {
     case "y":
       if (isComboChartConfig(chartConfig)) {
+        console.log(chartConfig);
```
Comment (bprusinowski) [inline app/configurator/components/chart-controls/control-tab.tsx]: 👁️

**Review by bprusinowski — COMMENTED** [review]
Nice changes! 👏 

I noticed two things when checking out the preview deployment:
- changing data filter reloads the whole page and "scrolls up", even when we didn't add a second cube (and we kind of lose track of the filter we changed and its new value),
- see https://visualization-tool-git-feat-add-dataset-ixt1.vercel.app/create/new?edit=aZzqcj_ca00d (log in as `ixt` user first) and remove the second cube – year key dimension is not selected as filter, and it should be 👀

**Comment by ptbrowne** [comment]
Thanks for this Bartosz, the flash of filters is due to previous data not being kept in the filters (the useQuery hook not returning the data if the key changes), previous data behavior should be tweaked for the filters. I'll do this in another PR. I'll also check the year filter not coming back there.

**Comment by ptbrowne** [comment]
In fact, I took a look and saw that if I added the keepPreviousData attribute handling (like in react-query), I could manage not to have the filters "reload" and show a loading indicator. The change of loading the multi-dataset data prior to actually changing the chart config must help there.

File: app/graphql/hooks.ts [hunk 4]
```diff
@@ -40,21 +46,23 @@ export const makeUseQuery =
...
+    const { keepPreviousData } = options;
     const executeQuery = React.useCallback(
       async (options: T) => {
         const result = await executeQueryFn(options.variables, () =>
           setResult((prev) => ({
             ...prev,
             fetching: true,
-            data: prev.queryKey === queryKey ? prev.data : null,
+            data:
+              prev.queryKey === queryKey || keepPreviousData ? prev.data : null,
```
Comment (bprusinowski) [inline app/graphql/hooks.ts:65]: 😍

**Review by bprusinowski — APPROVED** [review]
Thanks for the changes, the code looks 💯 :)

When testing the deployment preview, I've noticed one more thing which breaks the published chart when having two cubes merged 👀 

1. Go to https://visualization-tool-git-feat-add-dataset-ixt1.vercel.app/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/9&dataSource=Prod
2. Add "Greenhouse gas emissions by sector and by gas" dataset
3. See that Vertical Axis is empty (no label shown)
4. Publish the chart
5. See `No dimension <joinBy> in cube!` error

As these are not directly related to this PR (the same behavior on TEST), it's good to merge (but would be good to fix before INT deployment) 🙏 🤞

**Pull request merged**